src/traces/parcoords/attributes.js

```diff
@@ -0,0 +1,133 @@
+var extendFlat = require('../../lib/extend').extendFlat;
+
+module.exports = {
+
+    domain: {
+        x: {
+            valType: 'info_array',
+            role: 'info',
+            items: [
+                {valType: 'number', min: 0, max: 1},
+                {valType: 'number', min: 0, max: 1}
+            ],
+            dflt: [0, 1],
+            description: [
+                'Sets the horizontal domain of this `parcoords` trace',
+                '(in plot fraction).'
+            ].join(' ')
+        },
+        y: {
+            valType: 'info_array',
+            role: 'info',
+            items: [
+                {valType: 'number', min: 0, max: 1},
+                {valType: 'number', min: 0, max: 1}
+            ],
+            dflt: [0, 1],
+            description: [
+                'Sets the vertical domain of this `parcoords` trace',
+                '(in plot fraction).'
+            ].join(' ')
+        }
+    },
+
+    dimensions: {
+        _isLinkedToArray: 'dimension',
+        label: {
+            valType: 'string',
+            role: 'info',
+            description: 'The shown name of the dimension.'
+        },
+        tickvals: axesAttrs.tickvals,
+        ticktext: axesAttrs.ticktext,
+        visible: {
+            valType: 'boolean',
+            dflt: true,
+            role: 'info',
+            description: 'Shows the dimension when set to `true` (the default). Hides the dimension for `false`.'
+        },
+        range: {
+            valType: 'data_array',
+            role: 'info',
+            description: [
+                'The domain range that represents the full, shown axis extent. Defaults to the `values` extent.',
+                'Must be an array of `[fromValue, toValue]` with finite numbers as elements.'
+            ].join(' ')
+        },
+        constraintrange: {
+            valType: 'data_array',
+            role: 'info',
+            description: [
+                'The domain range to which the filter on the dimension is constrained. Must be an array',
+                'of `[fromValue, toValue]` with finite numbers as elements.'
+            ].join(' ')
+        },
+        values: {
+            valType: 'data_array',
+            role: 'info',
+            description: [
+                'Dimension values. `values[n]` represents the value of the `n`th point in the dataset,',
+                'therefore the `values` vector for all dimensions must be the same (longer vectors',
+                'will be truncated). Each value must be a finite number.'
+            ].join(' ')
+        },
+        description: 'The dimensions (variables) of the parallel coordinates chart. 2..60 dimensions are supported.'
+    },
+
+    pad: extendDeep({}, padAttrs, {
+        description: 'Set the padding of the `parcoords` component along each side to provide space for annotations.'
+    }, {t: {dflt: 80}, l: {dflt: 80}, r: {dflt: 80}, b: {dflt: 80}}),
```

Comment:
The direct answer is that the meaning of the current `parcoords` `pad` is the space to be left around the rectangular area that is the actual data projection place. The `parcoords` can be considered a row of contiguous panels of uniform height, onto which the lines are projected. It's not sufficient to show this only, because some annotations can (and do) get rendered along the edges, such as the dimension label on the top, the extreme values top/bottom, and tick and other values to the left and right. To retain these, and to avoid rendering to the very edge of the screen, there's a `pad` with quite considerable values. This way the layout is reasonable with just these defaults. It was originally named `padding` which I renamed after @etpinard's PR feedback, as `plotly` [already had](https://github.com/plotly/plotly.js/blob/master/src/plots/pad_attributes.js) `pad` with these exact properties.

As I believe you refer to the `margin` [block with l, r, b, t and pad]((https://plot.ly/javascript/setting-graph-size/)): It wasn't on the initial list of `layout` attributes to cover, and a quick test shows it doesn't have an effect. I see that there are _some_ `layout` attributes that have an effect even if I didn't do anything towards it, e.g. `paper_bgcolor`. It seems that 

- some of the `layout` attributes need implementation backing at the plot/trace level
- maybe I should cover more (or all?) `layout` attributes

Please let me know what kind of action is needed on this item.